### ISSUE TITLE
Support Python 3.12

### ISSUE THREAD

**Issue by esc** [numba/numba-scipy#101]
Seems like this package can not be built with 3.12 as of current `HEAD` at 1e2f24417d

https://github.com/numba/numba-scipy/pull/100#issuecomment-2042763667

**Comment by esc** [comment]
Looks like it might be versioneer update as per https://github.com/numba/numba-scipy/pull/75 for 3.11 ?

**Comment by RoyiAvital** [comment]
Is there any progress on this with support for the latest Numba (`0.6`) and SciPy (`1.14.1`)?

**Comment by esc** [comment]
> Is there any progress on this with support for the latest Numba (`0.6`) and SciPy (`1.14.1`)?

No, unfortunately not. None of the contributors has had time to continue work on this library. Pull-requests would likely be tended too however, so feel free to contribute!

**Comment by carlfischerjba** [comment]
FWIW it seems you can use numba-scipy on Python 3.12 with newer versions of numba and scipy without making any changes except for relaxing the version limits in `setup.py`. Tests pass as far as I can tell but CI isn't happy. I've been able to install direct from GitHub via my fork and all seems to work. I could put out a PR with those basic changes if it's appropriate but I'm unfamiliar with GitHub Actions and rusty when it comes to conda so might need some support.

**Comment by esc** [comment]
> FWIW it seems you can use numba-scipy on Python 3.12 with newer versions of numba and scipy without making any changes except for relaxing the version limits in `setup.py`. Tests pass as far as I can tell but CI isn't happy. I've been able to install direct from GitHub via my fork and all seems to work. I could put out a PR with those basic changes if it's appropriate but I'm unfamiliar with GitHub Actions and rusty when it comes to conda so might need some support.

The Numba team is very resource constrained, so apologies for the late response. If you have capacity, please do put in a PR to to update to 3.12 (maybe even 3.13) and ping me on the issue. If it's just fiddling with GHA I should be able to help.
